Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp

```diff
@@ -576,8 +560,21 @@ BytecodeGenerator::BytecodeGenerator(VM& vm, FunctionNode* functionNode, Unlinke
         // If we captured any formal parameter by name, then we use ScopedArguments. Otherwise we
         // use DirectArguments. With ScopedArguments, we lift all of our arguments into the
         // activation.
+        bool capturesAnyParameterByName = false;
+        if (functionNode->hasCapturedVariables()) {
+            for (size_t i = 0; i < parameters.size(); ++i) {
```

Comment:
We used to fetch `parameters` from `functionNode->parameters()`.  Why is it ok to use this version of parameters rather than th one from the `functionNode`?

Reply:
It's totally ok since the `parameters` is defined as

```cpp
FunctionParameters& parameters = *functionNode->parameters();
```

and it's not being mutated.